lib/hash.c

```diff
@@ -2263,7 +2263,7 @@ grn_hash_truncate(grn_ctx *ctx, grn_hash *hash)
   }
   key_size = hash->key_size;
   value_size = hash->value_size;
-  flags = hash->obj.header.flags;
+  flags = hash->header.common->flags;
```

Comment:
Do we need to preserve the existing hash->obj.header.flags, or is it okay to overwrite them?

Reply:
Made a call with @komainu8 
It’s fine to overwrite them.
The `hash->header.common->flags` already includes the existing information, so nothing will be lost.